### PR TITLE
Add an example incorporating templates

### PR DESCRIPTION
**Pull request by BenFradet**
That should help until we get #696

### CODE REVIEW

File: scio-examples/src/main/scala/com/spotify/scio/examples/extra/TemplateExample.scala [hunk 1]
```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2017 Spotify AB.
```
Comment (regadas) [inline scio-examples/src/main/scala/com/spotify/scio/examples/extra/TemplateExample.scala]: @BenFradet `2018`

File: scio-examples/src/test/scala/com/spotify/scio/examples/extra/TemplateExampleTest.scala [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2017 Spotify AB.
```
Comment (regadas) [inline scio-examples/src/test/scala/com/spotify/scio/examples/extra/TemplateExampleTest.scala]: @BenFradet `2018`

File: scio-examples/src/main/scala/com/spotify/scio/examples/extra/TemplateExample.scala [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      .as(classOf[Options])
+    options.setStreaming(true)
+    run(options)
+  }
+
+  def run(options: Options): Unit = {
+    val sc = ScioContext(options)
+
+    // We have to use a custom input to work with a ValueProvider
+    val input = sc
```
Comment (regadas) [inline scio-examples/src/main/scala/com/spotify/scio/examples/extra/TemplateExample.scala]: @BenFradet 🎨 I think it's cleaner if we keep fluent?

```scala
val inputIO = ...
val outputIO = ...

sc
  .customInput(....)
  .saveAsCustomOuput(...)
```

**Comment by BenFradet** [comment]
thanks, done :+1:

**Comment by regadas** [comment]
@BenFradet Thx!

**Pull request merged**